clash-lib/src/Clash/Primitives/Sized/Vector.hs

```diff
@@ -270,6 +271,13 @@ indexIntVerilog _isD _primName args _ty = return ((meta,) <$> bb)
     assign ~RESULT = ~SYM[0][~ARG[2]];~ELSEassign ~RESULT = ~ERRORO;~FI
     // index end|]
 
+  bbTextLitIx = [I.i|
+    // index lit begin
+    ~IF~SIZE[~TYP[1]]~THENwire ~TYP[1] ~GENSYM[vec][0] = ~ARG[1];
+    assign ~RESULT = ~SYM[0][~SIZE[~TYP[1]]-1-~LIT[2]*~SIZE[~TYPO] -: ~SIZE[~TYPO]];~ELSEassign ~RESULT = ~ERRORO;~FI
```

Comment:
```suggestion
    assign ~RESULT = ~SYM[0][~LENGTH[~TYP[1]]-1-~LIT[2]*~SIZE[~TYPO] -: ~SIZE[~TYPO]];~ELSEassign ~RESULT = ~ERRORO;~FI
```
I'm not exactly sure, but I assume you meant the length of the vector here @leonschoorl ? Because size will give you the number of bits.

Reply:
Ah, no, I got it now, `~SIZE` is correct.